Source/Core/DolphinQt/Debugger/WatchWidget.cpp

```diff
@@ -195,10 +195,11 @@ void WatchWidget::Update()
 
     QBrush brush = QPalette().brush(QPalette::Text);
 
-    if (!Core::IsRunning(m_system) || !PowerPC::MMU::HostIsRAMAddress(guard, entry.address))
+    const bool core_is_running = Core::IsRunning(m_system);
+    if (!core_is_running || !PowerPC::MMU::HostIsRAMAddress(guard, entry.address))
       brush.setColor(Qt::red);
 
-    if (Core::IsRunning(m_system))
+    if (core_is_running)
```

Comment:
Upon looking at this further, I would like to keep this as it is. While technically the first check makes the second check redundant, the first check is a workaround we added for performance reasons, whereas the second check is strictly required for correctness. The second check isn't exactly much extra complexity, so I think it would be good to keep it in case we figure out a way to get rid of the first check in the future. But if you disagree, feel free to open a PR.